rootfs/mkrootfs_debian.sh

```diff
@@ -26,11 +22,78 @@ function usage() {
 Build a Debian chroot filesystem image for testing libbbpf in a virtual machine.
 By default build an image for the architecture of the host running the script.
 
-    -a | --arch:    architecture to build the image for. Default (${arch})
+    -a | --arch:    architecture to build the image for. Default (${deb_arch})
     -d | --distro:  distribution to build. Default (${distro})
 "
 }
 
+function error() {
+    echo "ERROR: ${1}" >&2
+}
+
+function debian_to_gnu() {
+    # Funtion to convert an architecture in Debian to its GNU equivalent,
+    # e.g amd64 -> x86_64
+    # CPUTABLE contains a list of debian_arch\tgnu_arch per line
+    # Compare of the first field matches and print the second one.
+    awk -v deb_arch="$1" '$1 ~ deb_arch {print $2}' "${CPUTABLE}"
+}
+
+function qemu_static() {
+    # Given a Debian architecture find the location of the matching
+    # qemu-${gnu_arch}-static binary.
+    gnu_arch=$(debian_to_gnu "${1}")
+    echo "qemu-${gnu_arch}-static"
+}
+
+function check_requirements() {
+    # Checks that all necessary packages are installed on the system.
+    # print an error message explaining what is missing and exits.
+
+    local deb_arch=$1
+    local err=0
+
+    # Check that we can translate from Debian arch to GNU arch.
+    if [[ ! -e "${CPUTABLE}" ]]
+    then
+        error "${CPUTABLE} not found on your system. Make sure dpkg package is installed."
+        err=1
+    fi
+
+    # Check that the architecture is supported  by Debian.
+    if [[ -z $(debian_to_gnu "${deb_arch}") ]]
+    then
+        error "${deb_arch} is not a supported architecture."
+        err=1
+    fi
+
+    # Check that we can install the root image for a foreign arch.
+    qemu=$(qemu_static "${deb_arch}")
+    if [[ -z $(which "${qemu}") ]]
```

Comment:
I believe that `command -v` is the typical way for checking whether a command exists (https://manpages.org/command) [not that it will make a big difference]